crates/bevy_sprite/src/sprite.rs

```diff
@@ -1,16 +1,22 @@
+use bevy_asset::Handle;
 use bevy_color::Color;
 use bevy_ecs::{component::Component, reflect::ReflectComponent};
 use bevy_math::{Rect, Vec2};
 use bevy_reflect::{std_traits::ReflectDefault, Reflect};
+use bevy_render::{texture::Image, view::Visibility};
+use bevy_transform::components::Transform;
 
-use crate::TextureSlicer;
+use crate::{TextureAtlas, TextureSlicer};
 
-/// Specifies the rendering properties of a sprite.
-///
-/// This is commonly used as a component within [`SpriteBundle`](crate::bundle::SpriteBundle).
+/// Describes a sprite to be rendered to a 2D camera
 #[derive(Component, Debug, Default, Clone, Reflect)]
+#[require(Transform, Visibility)]
 #[reflect(Component, Default, Debug)]
 pub struct Sprite {
+    /// The image used to render the sprite
+    pub image: Handle<Image>,
+    /// The (optional) texture atlas used to render the sprite
+    pub atlas: Option<TextureAtlas>,
```

Comment:
As I understand it sprites are meant to be fairly fixed-function, so this change seems worth it for ergonomics and easier processing--many of the backend systems for sprites are now much simpler. I don't have a very strong opinion here myself but the [working group](https://hackmd.io/@bevy/required_components/%2FX6DrFv_fQm6J2VNEr4T4cQ) and @cart selected this option at least tentatively.

Reply:
I think the hackmd document could benefit for some rational of why a certain proposal was chosen.
I was expecting Texture atlas to stay a separate component (without any required component). Somewhat similar to how material does not require any components and mesh does not require a material (at least in the PR that was opened).

Reply:
Do you have a more concrete example somewhere that wouldn't be possible after this PR? I'm not super familiar with the current Sprite ecosystem and I have a hard time seeing how this change breaks third party crates, so an example would be useful for me.

Reply:
Thanks for the explanation, that makes sense. So basically if Sprite didn't have Image but was split into (Sprite, SpriteImage) then you can use the fact that replacing SpriteImage with CustomSpriteMaterial will prevent the builtin pipeline from running, allowing you to reuse Sprite.

While I wouldn't call that situation unusable (if you're setting up an entire pipeline then having a CustomSprite struct is a small extra cost), I can't really come up with a really good reason why we'd have them split for mesh2d and mesh3d and not here. I think cart or the rendering SMEs should weigh in on it to make sure we're doing this for good reasons.

Reply:
I'm not convinced we should compromise on the ergonomics of `Sprite` to support better customization. Afaict it's meant to be a fixed-function way to draw lots of quads *very quickly*. Custom materials compromises that. If it's the properties on the `Sprite` struct you want, why not just make a `MySprite` struct, copy the fields, and derive `Material2d`?. If I remember correctly there's been a bunch of recent optimizations to `Mesh2d`, which you likely wouldn't receive (maybe?) by manually overriding sprite rendering.

Reply:
if you have a bunch of these different cases, it might be nice to have `SpriteProperties` as a field of `Sprite`, `MySprite`, etc. Except that increases indenting for Sprite as well, so I'd say that should be on the user to implement (which would be fairly easy)

Reply:
resolving this thread in favor of continuing below